integration/upload_test_util.go

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2022 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+
+package integration
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/eclipse-kanto/file-upload/client"
+	"github.com/eclipse-kanto/kanto/integration/util"
+	"github.com/eclipse/ditto-clients-golang/protocol"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/net/websocket"
+)
+
+type uploadRequest struct {
+	CorrelationID string            `json:"correlationId"`
+	Options       map[string]string `json:"options"`
+}
+
+type uploadStatus struct {
+	State    string `json:"state"`
+	Progress int    `json:"progress"`
+}
+
+// ParseEventValue marshals an object(i.e map) and unmarshals it into a specific structure
+func ParseEventValue(props interface{}, result interface{}) error {
+	jsonValue, err := json.Marshal(props)
+	if err != nil {
+		return err
+	}
+	return json.Unmarshal(jsonValue, result)
+}
+
+// TriggerUploads executes an operation, which triggers file upload(s) and collects the upload requests
+func (suite *FileUploadSuite) TriggerUploads(featureID string, operation string, params interface{}, expectedFileCount int) map[string]string {
```

Comment:
The method is called `Trigger...` but the operation name is specified as an argument. Something is not correct here.